### PR TITLE
Re-implement the clone API with gRPC.

### PR DESCRIPTION
**Pull request by kfyharukz**
Signed-off-by: Kazuhito MATSUDA <kazuhito.matuda@gmail.com>

### CODE REVIEW

File: server/clone_grpc.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+		}
+
+		return &cloneParams{
+			donorHostName: donorHostName,
+			donorPort:     int(donorPort),
+			donorUser:     moco.CloneDonorUser,
+			donorPassword: agent.donorUserPassword,
+		}, nil
+	}
+
```
Comment (masa213f) [inline server/clone_grpc.go:148]: I think it is easier to understand to use `if else`.

File: server/clone_grpc_test.go [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+
+		for _, tt := range testcases {
+			By(tt.title)
+			_, err := gsrv.Clone(context.Background(), tt.req)
+			Expect(err).Should(HaveOccurred())
+		}
+
+		By("checking metrics")
+		// In these test cases, the clone will not start actually. So the metrics will not change.
+		_, err := getMetric(registry, metricsPrefix+"clone_count")
```
Comment (masa213f) [inline server/clone_grpc_test.go:109]: Why don't you check the value of the metrics?

Comment (masa213f) [inline server/clone_grpc_test.go:109]: Could you check the value of the `cluster_name` label?

Comment (kfyharukz) [reply server/clone_grpc_test.go:109]: I added some description to `getMetrics()`.

**Review by masa213f — APPROVED** [review]
LGTM

**Pull request merged**